R/autoplot.pop_data.R

```
@@ -6,142 +6,124 @@
-        ggplot2::labs(
-          title = "Distribution of Cross-sectional Antibody Responses (Log transformed)",
-          x = "Log10(Antibody Response Value)",
-          y = "Frequency"
-        )
-    } else
-    {
-      plot1 = plot1 +
-        ggplot2::labs(
-          title = "Distribution of Cross-sectional Antibody Responses",
-          x = "Antibody Response Value",
-          y = "Frequency"
-        )
-    }
-    return(plot1)
+      ggplot2::labs(
+        title = "Quantitative Antibody Responses by Age",
+        x = "Age",
+        y = "Value"
+      )
+  } else {
+    plot1 <- plot1 +
+      ggplot2::geom_point(
+        size = .6,
+        alpha = .7,
+        aes(color = get(strata))
+      ) +
+      ggplot2::geom_smooth(
+        method = lm,
+        se = FALSE,
+        aes(color = get(strata))
+      ) +
+      ggplot2::scale_y_log10() +
+      ggplot2::labs(
+        title = "Quantitative Antibody Responses by Age",
+        x = "Age",
+        y = "Value",
+        colour = strata
+      )
   }
 
-  # type error function
-  err_type = function()
-  {
-    return("Error:Select the correct type: {'density','age-scatter'}")
-  }
+  return(plot1)
+}
 
+# density plotting function
+density_plot <- function(
+    object,
+    strata = NULL,
+    log = FALSE) {
+  plot1 <-
+    object %>%
+    ggplot2::ggplot(aes(x = .data$value)) +
+    ggplot2::theme_linedraw() +
+    ggplot2::facet_wrap(~antigen_iso, nrow = 3)
 
-  if(type == 'age-scatter')
-  {
-    age_scatter(object,strata)
-  } else if (type == 'density')
-  {
-    density_plot(object,strata,log=FALSE)
-  } else
-  {
-    err_type()
+  if (is.null(strata)) {
+    plot1 <- plot1 +
+      ggplot2::geom_density(
+        alpha = .6,
+        color = "black"
+      )
+  } else {
+    plot1 <- plot1 +
+      ggplot2::geom_density(
+        alpha = .6,
+        color = "black",
+        aes(fill = get(strata))
+      ) +
+      ggplot2::labs(fill = strata)
   }
-
+  if (isTRUE(log)) {
```

Why do we need the `isTRUE()` function here? `log` should already be a `logical()` with length = 1; if the user gives us something else, we should throw an error, instead of interpreting it as FALSE

---

I kept getting this error `Error in if (log) { : argument is not interpretable as logical` while it is already set as `FALSE`. When I introduced 'isTRUE` the error disappeared.

---

I have managed to resolve the issue and remove `isTRUE`